File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/externaldms/update/ExternalDmsUpdate.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.update;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.id.IExternalDmsId;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.externaldms.update.ExternalDmsUpdate")
public class ExternalDmsUpdate implements IUpdate, IObjectUpdate<IExternalDmsId>
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IExternalDmsId externalDmsId;

    @JsonProperty
    private FieldUpdateValue<String> label = new FieldUpdateValue<String>();

    @JsonProperty
    private FieldUpdateValue<String> address = new FieldUpdateValue<String>();

    @Override
    @JsonIgnore
    public IExternalDmsId getObjectId()
    {
        return getExternalDmsId();
    }

    @JsonIgnore
    public IExternalDmsId getExternalDmsId()
    {
        return externalDmsId;
    }

    @JsonIgnore
    public void setExternalDmsId(IExternalDmsId externalDmsId)
    {
        this.externalDmsId = externalDmsId;
    }

    @JsonIgnore
    public FieldUpdateValue<String> getLabel()
    {
        return label;
    }

    @JsonIgnore
    public void setLabel(String label)
    {
        this.label.setValue(label);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getAddress()
    {
        return address;
    }

    @JsonIgnore
    public void setAddress(String address)
    {
        this.address.setValue(address);
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/externaldms/update/ExternalDmsUpdate.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/update/FieldUpdateValue" ], function(stjs, FieldUpdateValue) {
	var ExternalDmsUpdate = function() {
		this.label = new FieldUpdateValue();
		this.address = new FieldUpdateValue();
	};
	stjs.extend(ExternalDmsUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.externaldms.update.ExternalDmsUpdate';
		constructor.serialVersionUID = 1;
		prototype.externalDmsId = null;
		prototype.label = null;
		prototype.address = null;

		prototype.getObjectId = function() {
			return this.getExternalDmsId();
		};
		prototype.getExternalDmsId = function() {
			return this.externalDmsId;
		};
		prototype.setExternalDmsId = function(externalDmsId) {
			this.externalDmsId = externalDmsId;
		};
		prototype.getLabel = function() {
			return this.label;
		};
		prototype.setLabel = function(label) {
			this.label.setValue(label);
		};
		prototype.getAddress = function() {
			return this.address;
		};
		prototype.setAddress = function(address) {
			this.address.setValue(address);
		};
	}, {
		externalDmsId : "IExternalDmsId"
	});
	return ExternalDmsUpdate;
})